discovery.rst

```diff
@@ -15,14 +19,34 @@ The principle is always the same: you call the static ``find`` method on the dis
 implementation was specified. The discovery service will try to locate a suitable implementation.
 If no implementation is found, an ``Http\Discovery\NotFoundException`` is thrown.
 
-
 Installation
 ------------
 
+There are two kinds of installation:
+
+- In a reusable library
+- In an application
+
+In both cases you have to install the discovery package itself:
+
 .. code-block:: bash
 
         $ composer require php-http/discovery
 
+As mentioned above, discovery relies on Puli. In order to use it, you must set it up.
```

Comment:
it and it is confusing. "In order to use discovery, you need to also set up Puli."?